modules/users/server/controllers/users.profile.server.controller.js

```diff
@@ -301,33 +301,64 @@ exports.update = function (req, res) {
 
     // If user is changing email, check if it's available
     function (done) {
```

Comment:
@simison Since we have the same route for every profile update I'm not sure what would be the best way of handling this senario:

* The user has used her/his change of changing the username already so every new try the server will return an error saying it is not possible.

The user has now 2 input boxes. The first one is to change the username and the second one to change the e-mail address. She/He has also two buttons to save the data.

Now let's do this: The user changes the input box related to the username to a different username but the user does not click on save. Now the same user goes to the second input box and change the e-mail to a new one. Because we use the same function to update the profile. The user will receive a message saying it can't update the username :(

Like I said, I'm new to Javascript and new to trustroots so I might be leaving things behind...
sooo...Shall we have a different route to update the username? What would it be the best solution?

Reply:
The client should handle these situations separately. I.e. clicking the `Change Username` button should send the PUT request to the API with edited username only. The same with email.